src/main/java/net/fabricmc/tinyremapper/api/WrapperFunction.java

```diff
@@ -0,0 +1,10 @@
+package net.fabricmc.tinyremapper.api;
+
+import org.objectweb.asm.ClassVisitor;
+
+/**
+ * @see ClassVisitor#ClassVisitor(int, ClassVisitor)
+ */
+public interface WrapperFunction {
+	ClassVisitor wrap(ClassVisitor visitor, ExtendedRemapper remapper, Classpath classpath);
```

Comment:
Maybe also consider adding @FunctionalInterface for this interface.

Reply:
Must the visitor propagate at real time or can it cache the results and mass dump calls later? Such as we first grab everything from the source visitor to a class node, do stuff to the node, and finally we dump a visitor to the node, instead of delegating most calls and only patching a few selected calls.